src/openms/source/CHEMISTRY/AASequence.cpp

```diff
@@ -107,6 +107,68 @@ namespace OpenMS
     return tmp;
   }
 
+  String AASequence::toUniModString() const
+  {
+    const AASequence & seq = *this;
+
+    String bs;
+    int unimod = -1;
+
+    if (seq.empty()) return bs;
+
+    if (seq.hasNTerminalModification())
+    {
+      const ResidueModification& mod = *(seq.getNTerminalModification());
+      unimod = mod.getUniModRecordId();
+      if (unimod > -1)
+      {
+        bs += ".(UniMod:" + String(unimod) + ")";
+      }
+      else
+      {
+        bs += ".[" + String(mod.getDiffMonoMass()) + "]";
+      }
+    }
+
+    for (Size i = 0; i != seq.size(); ++i)
+    {
+      const Residue& r = seq[i];
+      const String aa = r.getOneLetterCode() != "" ? r.getOneLetterCode() : "X";
```

Comment:
just related to this PR: in which cases do we expect this to be empty? This is somewhat poorly documented within Residue class.